test/framework/src/main/java/org/elasticsearch/index/KnownIndexVersions.java

```diff
@@ -19,4 +19,6 @@ public class KnownIndexVersions {
      * A sorted list of all known transport versions
      */
     public static final List<IndexVersion> ALL_VERSIONS = List.copyOf(IndexVersions.getAllVersions());
+
+    public static final List<IndexVersion> ALL_READ_ONLY_VERSIONS = List.copyOf(IndexVersions.getAllReadOnlyVersions());
```

Comment:
I'm not entirely sure why we need this list copy but since thats the pattern for ALL_VERSIONS I thought its save to do the same. Maybe someone know why we need that for ALL_VERSIONS?

Reply:
By checking its history a bit, I think, the initial intention was to make a defensive copy of the list, which is immutable in the current version. The list was also unmodifiable when it was first introduced: https://github.com/elastic/elasticsearch/pull/93384

Reply:
I wonder if we need to have a list with read only versions. Could we instead just add a new method to get read only versions? Also, should we instead have a new method that returns all versions, writeable and read-only, as opposed to having special tests for read-only indices? My expectations it that the majority of the cases will need read support, and no write, hence the special case should probably be for writing and not for reading.